### DISCUSSION TITLE
Extending existing units with per and times methods

### DISCUSSION THREAD

**Discussion by narmstro2020** [wpilibsuite/2025Beta#57] [Q&A]
Playing around with creating my own units for various projects.  

I'd love to extend the existing usage of something like Torque or Angular velocity by adding my own per and times methods that will return my new Units class.  Unfortunately the built in methods are final.  Is there a way around it.

**Comment by SamCarlberg** [comment]
@narmstro2020 The unit classes could be made non-final. There may be some friction with generic type bounds, which is why those classes were made final in the first place

**Reply by narmstro2020** [reply]
If anything the extensions I had in mind for my local code would just solidify the generic type bounds with an existing type.

**Reply by narmstro2020** [reply]
If only Java supported extension method
